zilliqa/src/block_store.rs

```diff
@@ -241,6 +261,10 @@ impl BlockStore {
         self.get_block(hash)
     }
 
+    pub fn get_highest_block_number(&self) -> Result<Option<u64>> {
```

Comment:
Can you replace the uses of `db.get_highest_block_number()` with `block_store.get_highest_block_number()` if we are adding this. Some are withing `BlockStore` itself, but I guess we have some in `Consensus` too.